### PR TITLE
Changed the parameter WebsiteName to Name to use in piping. Added publis...

### PR DESCRIPTION
**Pull request by NaveenGoli**
Added Publishing profile and changed the websitename to name in the parameters.

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@NaveenGoli__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (ngoli). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

**Comment by AzureRT** [comment]
Can one of the admins verify this patch?

File: src/ResourceManager/Websites/Commands.Websites/Models.Websites/WebsiteBaseSlotCmdlet.cs [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Websites.Models;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Websites.Models.Websites;
+
+namespace Microsoft.Azure.Commands.Websites
+{
+    public class WebsiteBaseSlotCmdlet : WebsitesBaseClient
```
Comment (amitapl) [inline src/ResourceManager/Websites/Commands.Websites/Models.Websites/WebsiteBaseSlotCmdlet.cs]: should derive from WebsiteBaseCmdlet

File: src/ResourceManager/Websites/Commands.Websites/Utilities/WebsitesClient.cs [hunk 2]
```diff
@@ -92,58 +92,79 @@ public System.Net.HttpStatusCode RemoveWebsite(string resourceGroupName, string
...
             WebSiteGetParameters webSiteGetParams = new WebSiteGetParameters();
-
             var getWebsite = WrappedWebsitesClient.WebSites.Get(resourceGroupName, webSiteName, slotName, webSiteGetParams);
+            try
+            {
+                var getAppSettings = WrappedWebsitesClient.WebSites.GetAppSettings(resourceGroupName, webSiteName, slotName);
+                //Add websiteApp Settings to the Website object as the Get call will not return them.
+                foreach (var appSettingVal in getAppSettings.Resource.Properties.ToList())
+                {
+                    if (!getWebsite.WebSite.Properties.Properties.AppSettings.Keys.Contains(appSettingVal.Name))
```
Comment (amitapl) [inline src/ResourceManager/Websites/Commands.Websites/Utilities/WebsitesClient.cs]: why do you need this check?

Comment (amitapl) [inline src/ResourceManager/Websites/Commands.Websites/Utilities/WebsitesClient.cs]: also if you do its missing {}

File: src/ResourceManager/Websites/Commands.Websites/Utilities/WebsitesClient.cs [hunk 3]
```diff
@@ -92,58 +92,79 @@ public System.Net.HttpStatusCode RemoveWebsite(string resourceGroupName, string
...
+            catch
+            {
+                //ignore if this call fails as it will for reader RBAC
+            }
+            //Add ConnectionStrings Settings to the Website object as the Get call will not return them.
+            
+            try
+            {
+                var getConnectionStringsSettings = WrappedWebsitesClient.WebSites.GetConnectionStrings(resourceGroupName, webSiteName, slotName);
+                //TODO: Add ConnectionStrings Settings to the Website object as the Get call will not return them.
```
Comment (amitapl) [inline src/ResourceManager/Websites/Commands.Websites/Utilities/WebsitesClient.cs]: you should either use `getConnectionStringsSettings ` or remove this code

File: src/ResourceManager/Websites/Commands.Websites/Models.Websites/WebsiteBaseNotMandatoryCmdlet.cs [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Websites.Models;
+using System.Management.Automation;
+using Microsoft.Azure.Commands.Websites.Models.Websites;
+
+namespace Microsoft.Azure.Commands.Websites
+{
+    public class WebsiteBaseNotMandatoryCmdlet : WebsitesBaseClient
```
Comment (amitapl) [inline src/ResourceManager/Websites/Commands.Websites/Models.Websites/WebsiteBaseNotMandatoryCmdlet.cs]: not really related to this change but I think that `WebsitesBaseClient` should be renamed as it's misleading, it is actually a Cmdlet and not a Client (only holds a client).

**Comment by amitapl** [comment]
Looks good, minor comments

File: src/ResourceManager/Websites/Commands.Websites/Utilities/WebsitesClient.cs (hunk 3 above)
Comment (NaveenGoli) [inline src/ResourceManager/Websites/Commands.Websites/Utilities/WebsitesClient.cs]: its in TODO: waiting on a modified website object will update tomorrow.

File: src/ResourceManager/Websites/Commands.Websites/Models.Websites/WebsiteBaseSlotCmdlet.cs (hunk 1 above)
Comment (NaveenGoli) [inline src/ResourceManager/Websites/Commands.Websites/Models.Websites/WebsiteBaseSlotCmdlet.cs]: Changed this with other changes in the pipe will update in next pull

File: src/ResourceManager/Websites/Commands.Websites/Utilities/WebsitesClient.cs (hunk 2 above)
Comment (amitapl) [inline src/ResourceManager/Websites/Commands.Websites/Utilities/WebsitesClient.cs]: `getWebsite.WebSite.Properties.Properties.AppSettings.Add[appSettingVal.Name] = appSettingVal.Value`

**Comment by amitapl** [comment]
:shipit:

**Pull request merged**